core/src/main/kotlin/com/strumenta/kolasu/parsing/KolasuParser.kt

```diff
@@ -60,26 +60,10 @@ abstract class KolasuParser<R : Node, P : Parser, C : ParserRuleContext> : ASTPa
         issues: MutableList<Issue>
     ): R?
 
-    /**
-     * Performs "lexing" on the given code string, i.e., it breaks it into tokens.
-     */
-    @JvmOverloads
-    fun lex(code: String, onlyFromDefaultChannel: Boolean = true): LexingResult {
-        val charset = Charsets.UTF_8
-        return lex(code.byteInputStream(charset), charset, onlyFromDefaultChannel)
-    }
-
-    /**
-     * Performs "lexing" on the given code stream, i.e., it breaks it into tokens.
-     */
-    @JvmOverloads
-    fun lex(
-        inputStream: InputStream,
-        charset: Charset = Charsets.UTF_8,
-        onlyFromDefaultChannel: Boolean = true
-    ): LexingResult {
+    override fun lex(inputStream: InputStream, charset: Charset, onlyFromDefaultChannel: Boolean): LexingResult {
```

Comment:
It seems to me that we lost the default values, the comment, and the @JvmOverloads annotation (useful for using this from Java)

Reply:
I moved those to the interface.
Unfortunately, in Kotlin, interfaces can't have `@JvmOverloads`, and implementation cannot either, so I had to manually add all the overloads.

Reply:
But maybe this is the case as we are overriding?